Editor/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,286 @@
+            if (vcamBase == null) continue;
+
+            var vcam = vcamBase as CinemachineVirtualCamera;
+            if (vcam != null) 
+            {
+                vcamDatas.Add(ConvertVcamToVcamData(vcam, i.ToString()));
+            }
+            else // Composite vcam (Freelook, Mixing, Statedriven, Clearshot):
+            {
+                GetExtensions(vcamBase, out List<string> vcamExtensions, out int customExtensionCount);
+
+                string id = i.ToString();
+                vcamDatas.Add(new VcamData
+                {
+                    id = id,
+                    vcam_class = GetVcamClassName(vcamBase),
+                    has_follow_target = vcamBase.Follow != null,
+                    has_lookat_target = vcamBase.LookAt != null,
+                    blend_hint = "",
+                    inherit_position = false,
+                    stand_by_update = vcamBase.m_StandbyUpdate.ToString(),
+                    mode_overwrite = "",
+                    body_component = "",
+                    aim_component = "",
+                    noise_component = "",
+                    custom_component_count = 0,
+                    extensions = vcamExtensions.ToArray(),
+                    custom_extension_count = customExtensionCount,
+                });
+
+                var vcamChildren = vcamBase.GetComponentsInChildren<CinemachineVirtualCamera>();
+                for (var c = 0; c < vcamChildren.Length; c++)
+                {
+                    vcamDatas.Add(ConvertVcamToVcamData(vcamChildren[c], id + "." + c));
+                }
+            }
+        }
+        
+        var projectData = new ProjectData
+        {
+            brain_count = cinemachineCore.BrainCount,
+            vcam_count = cinemachineCore.VirtualCameraCount,
+            cam_count = Camera.allCamerasCount,
+            vcams = vcamDatas,
+            time_elapsed = Time.realtimeSinceStartup - startTime,
+        };
+        
+        // Register our event like this
+        EditorAnalytics.RegisterEventWithLimit("cm_vcams_on_play", k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+
+        // Send the data to the database
+        EditorAnalytics.SendEventWithLimit("cm_vcams_on_play", projectData);
+    }
+    
+    [Serializable]
+    private struct ProjectData
+    {
+        public int brain_count;
+        public int vcam_count;
+        public int cam_count;
+        public List<VcamData> vcams;
+        public float time_elapsed;
+    }
+
+    [Serializable]
+    private struct VcamData
+    {
+        public string id;
+        public string vcam_class;
+        public bool has_follow_target;
+        public bool has_lookat_target;
+        public string blend_hint;
+        public bool inherit_position;
+        public string stand_by_update;
+        public string mode_overwrite;
+        public string body_component;
+        public string aim_component;
+        public string noise_component;
```

Comment:
I don't see the point of having 3 separate fields here.  Why not just a single components[] field, similar to extensions[] ?

Reply:
I think, the reasoning was that we'd like to know if the user created a custom body, aim or noise component.

Reply:
if you make a single component array (like extensions), then then entry for a custom component can be something like "custom_aim", "custom_body), etc.  You can build the string by doing something like "custom_" + component.Stage.ToString()

Reply:
The other reason was to not have an array, because it makes the queries slow. Because we know there can be only max 3 components, it is better to have 3 fields separately.
So we agreed to have 3 fields for components instead of an array of components.